src/Kiota.Builder/Writers/CSharp/CodeMethodWriter.cs

```diff
@@ -203,6 +204,15 @@ private void WriteFactoryMethodBody(CodeMethod codeElement, CodeClass parentClas
             WriteFactoryMethodBodyForUnionModel(codeElement, parentClass, parseNodeParameter, writer);
         else if (parentClass.DiscriminatorInformation.ShouldWriteDiscriminatorForIntersectionType)
             WriteFactoryMethodBodyForIntersectionModel(codeElement, parentClass, parseNodeParameter, writer);
+        else if (codeElement.Name == "CreateFromDiscriminatorValueWithMessage" && parentClass.IsErrorDefinition)
+        {
+            // Special case: CreateFromDiscriminatorValueWithMessage for error classes
+            var messageParam = codeElement.Parameters.FirstOrDefault(p => p.Name == "message");
```

Comment:
rather than selection on the name, could we use a kind instead? it's ok to introduce a new kind

Reply:
That makes the code a lot cleaner and easier to follow, I'll do that.